Rpackage/R/growtree.R

```diff
@@ -1,38 +1,49 @@
-growtree <- structure(function(target.mat, feature.mat, depth=0, maxdepth = Inf,
+growtree <- structure(function(target.mat, feature.mat, depth = 0, max_depth = Inf,
                                margin = 0.0, loss="hinge", id = 1L, min_sample = 1,
                                pred=NULL, side = NULL, weights = NULL) {
   
   node_info_print <- NULL
   ### creating dataframe of info
   if(!is.null(pred)){
-    node_info_print <- data.frame(pred[[side]], pred[[1]],row.names = "")
-    colnames(node_info_print) <- c("prediction", "cost")
+    ###left
+    if(side == 1){
+      node_info_print <- data.frame(pred$leftpred, pred$leftcost, row.names = "")
+      colnames(node_info_print) <- c("prediction", "cost")
+    }
+    else{
+      node_info_print <- data.frame(pred$rightpred, pred$rightcost, row.names = "")
+      colnames(node_info_print) <- c("prediction", "cost")
+    }
+    
   }
   
   ### if node is root.
   if(is.null(side)){
     ### calculate root node cost (as no split thus leftcost + rightcost = 0 + cost )
     node_info_print <- compute_optimal_costs(target.mat, margin, loss)
     node_info_print <- cbind(node_info_print[[2]], node_info_print[[3]])
-    node_info_print <- node_info_print[length(node_info_print[,1]),]
+    node_info_print <- node_info_print[nrow(node_info_print),]
     node_info_print <- data.frame(node_info_print[1], node_info_print[2], row.names = "")
     colnames(node_info_print) <- c("prediction", "cost")
   }
   
   ### if depth exceeds we stop the further growth of tree
-  if (depth >= maxdepth) return(partynode(id = id, info = node_info_print))
+  if (depth >= max_depth) return(partynode(id = id, info = node_info_print))
   
 
   ### if sample per node is less than minimum, we stop tree growth
   if(sum(weights) <= min_sample) return(partynode(id = id, info = node_info_print))
 
   ### split the tree at the node.
-  sp <- bestsplit(target.mat, feature.mat, weights, margin, loss, pred)
+  sp <- bestsplit(target.mat, feature.mat, weights, margin, loss, node_info_print)
   splt <- sp
-
+  
   ### if no split, we stop
   if (is.null(sp)) return(partynode(id = id, info = node_info_print))
-
+  
+  ### tree cant grows if splitting cost is more than current node cost
+  assert_that(node_info_print$cost >= (sp$leftcost + sp$rightcost))
```

Comment:
Equal should not be in this condition since if its equal then you're better off not splitting at all.